packages/contracts-bedrock/src/L1/L2OutputOracle.sol

```diff
@@ -300,4 +308,12 @@ contract L2OutputOracle is Initializable, ISemver {
     function computeL2Timestamp(uint256 _l2BlockNumber) public view returns (uint256) {
         return startingTimestamp + ((_l2BlockNumber - startingBlockNumber) * L2_BLOCK_TIME);
     }
+
+    /// @notice Returns a boolean indicating whether the given address can delete an output.
+    /// @param _addr The address to check.
+    /// @return True if the address can delete an output, false otherwise.
+    function mayDelete(address _addr) public view returns (bool) {
+        return _addr == systemConfig.challenger() || _addr == systemConfig.superchainConfig().initiator()
```

Comment:
This does extra calls that are not necessary

Reply:
I think this is the best way to do it, here is my reasoning:

1. The addresses being retrieved should be stored in as few locations as possible: 
	- systemConfig in the L2OutputOracle
	- superchainConfig in the SystemConfig
	- initiator and vetoer in the SuperChainConfig
1. Based on the above assertion, the only optimization that I can see would be to add a getter to the SuperChainConfig which returns both the initiator and vetoer. IMO the additional code maintenance ($100/loc), is not justified, because:
1. This method will only be executed on-chain when an output is being deleted, so it is not a priority for optimization.

Reply:
What i mean is that you call `systemConfig.superchainConfig()` twice

Reply:
oh lol, fair point

Reply:
If you feel strongly about it i'm fine with leaving it but there really isn't a reason to call `systemConfig.superchainConfig()` twice

```solidity
SuperchainConfig superchainConfig = systemConfig.superchainConfig();
return addr == systemConfig.challenger() || _addr == superchainConfig().initiator() || _addr == superchainConfig().vetoer();
```